core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -306,18 +428,19 @@ class LongAccumulator extends AccumulatorV2[jl.Long, jl.Long] {
 
   /**
    * Adds v to the accumulator, i.e. increment sum by v and count by 1.
-   * @since 2.0.0
+   * Added for simplicity with adding non java Longs.
+   * @since 2.1.0
    */
-  override def add(v: jl.Long): Unit = {
-    _sum += v
-    _count += 1
+  def add(v: Long): Unit = {
```

Comment:
yeah I don't love the copied code either.  The thing is, `@specialized` has bigger ramfications -- the actual classes that get used (eg. that you see in stack traces) are different,  tricker to access the right class from java, there are a ton of generated classes when there is multiple inheritance (not relevant here at the moment, but perhaps in the future).  I think having copied code is the lesser of two evils.